src/OpenTelemetry.AutoInstrumentation/Instrumentations/NLog/Bridge/Integrations/LoggerIntegration.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+using OpenTelemetry.AutoInstrumentation.Logging;
+#if NET
+using OpenTelemetry.AutoInstrumentation.Logger;
+#endif
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.Bridge.Integrations;
+
+/// <summary>
+/// NLog Logger integration that hooks into the actual logging process.
+/// This integration intercepts NLog's Logger.Log method calls to automatically
+/// capture log events and forward them to OpenTelemetry when the NLog bridge is enabled.
+///
+/// The integration targets NLog.Logger.Log method which is the core method called
+/// for all logging operations, allowing us to capture events without modifying configuration.
+/// </summary>
+[InstrumentMethod(
+assemblyName: "NLog",
+typeName: "NLog.Logger",
+methodName: "Log",
+returnTypeName: ClrNames.Void,
+parameterTypeNames: new[] { "NLog.LogEventInfo" },
+minimumVersion: "4.0.0",
+maximumVersion: "6.*.*",
+integrationName: "NLog",
+type: InstrumentationType.Log)]
+public static class LoggerIntegration
```

Comment:
@zacharycmontoya, could you please advice if it is the best integration point for this behavior?
As I remember DD supports NLog bridges/context injection.